src/node_sqlite.cc

```diff
@@ -585,27 +589,98 @@ bool DatabaseSync::ShouldIgnoreSQLiteError() {
+  if (!path->IsString()) {
+    return false;
+  }
+
+  *location = Utf8Value(env->isolate(), path.As<String>()).ToString();
+  return true;
+}
+
+bool HandleDatabaseLocationFromBuffer(Environment* env,
+                                      Local<Value> path,
+                                      std::string* location) {
+  if (!path->IsUint8Array()) {
+    return false;
+  }
+
+  Local<Uint8Array> buffer = path.As<Uint8Array>();
+  size_t byteOffset = buffer->ByteOffset();
+  size_t byteLength = buffer->ByteLength();
+  auto data =
+      static_cast<const uint8_t*>(buffer->Buffer()->Data()) + byteOffset;
+  if (std::find(data, data + byteLength, 0) != data + byteLength) {
+    return false;
+  }
+
+  Local<Value> out;
+  if (String::NewFromUtf8(env->isolate(),
+                          reinterpret_cast<const char*>(data),
+                          NewStringType::kNormal,
+                          static_cast<int>(byteLength))
+          .ToLocal(&out)) {
+    *location = Utf8Value(env->isolate(), out.As<String>()).ToString();
+    return true;
+  }
+
+  return false;
+}
+
+bool HandleDatabaseLocationFromURL(Environment* env,
+                                   Local<Value> path,
+                                   std::string* location,
+                                   std::string* protocol_v) {
+  if (!path->IsObject()) {
+    return false;
+  }
+
+  Local<Object> url = path.As<Object>();
+  Local<Value> href;
+  Local<Value> protocol;
+  if (!url->Get(env->context(), env->href_string()).ToLocal(&href) ||
+      !href->IsString() ||
+      !url->Get(env->context(), env->protocol_string()).ToLocal(&protocol) ||
+      !protocol->IsString()) {
+    return false;
+  }
+
+  *protocol_v = Utf8Value(env->isolate(), protocol.As<String>()).ToString();
+  *location = Utf8Value(env->isolate(), href.As<String>()).ToString();
+  return true;
+}
+
 void DatabaseSync::New(const FunctionCallbackInfo<Value>& args) {
   Environment* env = Environment::GetCurrent(args);
-
   if (!args.IsConstructCall()) {
     THROW_ERR_CONSTRUCT_CALL_REQUIRED(env);
     return;
   }
 
-  if (!args[0]->IsString()) {
+  std::string location;
+  std::string url_protocol;
+  Local<Value> path = args[0];
+  bool valid_path =
+      HandleDatabaseLocationFromString(env, path, &location) ||
+      HandleDatabaseLocationFromBuffer(env, path, &location) ||
+      HandleDatabaseLocationFromURL(env, path, &location, &url_protocol);
+  if (!valid_path) {
     THROW_ERR_INVALID_ARG_TYPE(env->isolate(),
-                               "The \"path\" argument must be a string.");
+                               "The \"path\" argument must be a string, "
+                               "Uint8Array, or URL without null bytes.");
     return;
   }
 
-  std::string location =
-      Utf8Value(env->isolate(), args[0].As<String>()).ToString();
-  DatabaseOpenConfiguration open_config(std::move(location));
+  if (url_protocol != "" && url_protocol != "file:") {
```

Comment:
I think this should be moved into the location validation function, and only run when the input is actually a URL.